src/index.js

```diff
@@ -5,46 +5,56 @@ import kafka from 'no-kafka';
 import NULL_LOGGER from 'null-logger';
 import Task from './task';
 import Registry from './registry';
-import Runner from './runner';
-import Admin from './admin';
-import Producer from './producer';
+import runner from './base/runner';
+import admin from './base/admin';
+import producer from './base/producer';
 import Config from './config';
 
-import type { Callback, Configuration, KafkaCompression } from '../types';
-
-const Steveo = (configuration: Configuration, logger: Object = NULL_LOGGER) => () => {
-  const registry = Registry();
-  const config = new Config(configuration);
-  let getTopicName = null;
-
-  const task = (topic: string, callBack: Callback) => {
-    const producer = Producer(config, registry, logger);
+import type { ITask, Configuration, Callback, Logger, ISteveo, IRegistry, IEvent, IAdmin } from '../types';
+
+class Steveo implements ISteveo {
+  config: Configuration;
+  logger: Logger;
+  registry: IRegistry;
+  getTopicName: Callback;
+  admin: IAdmin;
+  events: IEvent;
+
+  constructor(configuration: Configuration, logger :Logger = NULL_LOGGER) {
+    this.logger = logger;
+    this.registry = new Registry();
+    this.config = new Config(configuration);
```

Comment:
@davidbanham yeah, the idea is to publish `steveo` with support for multiple backends. As you suggested, we will also plan for a selective installation `steveo/kafka`. After refactoring the backend code is abstracted out.Tthe main difference is only in runner & producer while reusing task & registry.